src/dev/mocha/junit_report_generation.js

```diff
@@ -23,6 +23,40 @@ import { inspect } from 'util';
 
 import mkdirp from 'mkdirp';
 import xmlBuilder from 'xmlbuilder';
+import stripAnsi from 'strip-ansi';
+import regenerate from 'regenerate';
+
+import { getSnapshotOfRunnableLogs } from './log_cache';
+
+// create a regular expression using regenerate() that selects any character that is explicitly allowed by https://www.w3.org/TR/xml/#NT-Char
+const validXmlCharsRE = new RegExp(
+  `(?:${
+    regenerate()
+      .add(0x9, 0xA, 0xD)
+      .addRange(0x20, 0xD7FF)
+      .addRange(0xE000, 0xFFFD)
+      .addRange(0x10000, 0x10FFFF)
+      .toString()
+  })+`,
+  'g'
+);
+
+function escapeCdata(string) {
+  const colorless = stripAnsi(string);
+  let validXmlChars = '';
+  while (true) {
```

Comment:
__question:__ I'm curious how's this `exec`-in-a-loop different from this:
```js
return stripAnsi(string).match(validXmlCharsRE).join('')
```

Reply:
Huh, I guess it's not, and that is much cleaner.